src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -23,38 +37,189 @@
+  private NetworkTableEntry DShooterRPM = shooterTab.add("Shooter RPM", 0.0).withPosition(2,1).getEntry();
+  private NetworkTableEntry BCargoRunning = shooterTab.add("Is the CDS Running",false).withPosition(2, 2).getEntry();
+  private NetworkTableEntry DShooterRPMInput = shooterTab.add("Shooter RPM Input",2000).withPosition(2, 3).getEntry();
+  private NetworkTableEntry IDelayTable = shooterTab.add("Current I delay",0).withPosition(2, 4).getEntry(); //Delay before the CDS deliver the ball for the PID to stablize the speed
+  private double MaxOutputConstant;
+  private double MinOutputConstant;
+  private ShooterConfig[] DistanceArray;
+
+
+
+
+
+  
+  
+  //private NetworkTableEntry PID_F = shooterTab.addPersistent("PID F", Constants.Shooter.).withPosition(3, 2).getEntry();
+  //private NetworkTableEntry PID_Izone = shooterTab.addPersistent("PID I Range", Constants.Shooter.).withPosition(4, 2).getEntry();
+  //private NetworkTableEntry PID_MaxOutput = shooterTab.addPersistent("PID Peak Output", Constants.Shooter.d).withPosition(5, 2).getEntry();
+
+
+
+
+
+
+
+
   public ShooterSubsystem() {
-    
+    //SmartDashboard.putNumber("RPMIN", RPMIN);
     aimMode = 4;
-    cargo_motorController = new MotorController("Shooter Cargo", Constants.shooterCargoID);
-    shooter_motorController = new MotorController("Shooter", Constants.shooterID, 40, true);
+    cargo_motorController = new MotorController("Shooter Cargo", Constants.Shooter.shooterCargoID,40,true);
+    kCargoController = cargo_motorController.getPID();
+    kCargoEncoder = cargo_motorController.getEncoder();
+    shooter_motorController = new MotorController("Shooter", Constants.Shooter.shooterID, 40, true);
     KShooterController = shooter_motorController.getPID();
     KShooterEncoder = shooter_motorController.getEncoder();
-    KShooterController.setP(5e-4);
-    KShooterController.setI(6e-7);
-    KShooterController.setD(0.0);
+    
+    DistanceArray[0] = new ShooterConfig(0,0, 0);
+    DistanceArray[1] = new ShooterConfig(0, 0, 0);
+    DistanceArray[2] = new ShooterConfig(0,0, 0);
+    DistanceArray[3] = new ShooterConfig(0,0, 0);
+    DistanceArray[4] = new ShooterConfig(0,0, 0);
+    DistanceArray[5] = new ShooterConfig(0,0, 0);
+    DistanceArray[6] = new ShooterConfig(0,0, 0);
+    DistanceArray[7] = new ShooterConfig(0,0, 0);
+    DistanceArray[8] = new ShooterConfig(0,0, 0);
+    DistanceArray[9] = new ShooterConfig(0,0, 0);
+    DistanceArray[10] = new ShooterConfig(0,0, 0);
+    DistanceArray[11] = new ShooterConfig(0,0, 0);
+    DistanceArray[12] = new ShooterConfig(0,0, 0);
+    DistanceArray[13] = new ShooterConfig(0,0, 0);
+    DistanceArray[14] = new ShooterConfig(0,0, 0);
+    DistanceArray[15] = new ShooterConfig(0,0, 0);
+    DistanceArray[16] = new ShooterConfig(0,0, 0);
+    DistanceArray[17] = new ShooterConfig(0,0, 0);
+    DistanceArray[18] = new ShooterConfig(0,0, 0);
+    DistanceArray[19] = new ShooterConfig(0,0, 0);
```

Comment:
These should be static and outside of the constructor.  They won't change in runtime.

Only have workable values in the list.  It doesn't help to have 20 0,0,0 values. 
You can't test with this at all.

Make up three reasonable test values, and start with that, and fill out more.

You don't need to specify the index when you are creating an array of objects.

Example here:  https://github.com/caseyjbrotherton/testLookup/blob/main/src/main/java/austincans/TestLookup.java#L105-L114

Need to make sure that they are sorted.  If you aren't explicitly sorting them in code, then note the requirement in comments.